addons/stock/views/stock_rule_views.xml

```diff
@@ -88,7 +89,8 @@
                             <group name="apply_on" string="Applicability">
                                 <field name="route_id"/>
                                 <field name="warehouse_id" attrs="{'invisible': [('action', '=', 'push')]}" groups="base.group_no_one"/>
-                                <field name="company_id" attrs="{'required': [('action', '=', 'push')]}" groups="base.group_multi_company"/>
+                                <field name="route_company_id" invisible="1"/>
+                                <field name="company_id" attrs="{'required': [('action', '=', 'push')]}" domain="[('id', '=?', route_company_id)]" groups="base.group_multi_company"/>
```

Comment:
does it work?
we want to be able to set any company on a rule if no company is set on a route
here it seems if no company is set on the route you can't set a company on the rule?